src/dd/dd.rs

```diff
@@ -0,0 +1,640 @@
+            of_ = self.out_name,
+            ibs = self.inp_block_size,
+            obs = self.out_block_size,
+            cbs = self.cnv_block_size,
+            skip = self.skip,
+            seek = self.seek,
+            count = self.count,
+            flags = self.str_flags()
+        )
+    }
+}
+
+pub fn parse_args() -> Conf {
+    let mut cf = Conf::default();
+    for arg in env::args().skip(1) {
+        let pair: Vec<&str> = arg.splitn(2, '=').collect();
+        if pair.len() == 2 {
+            let (key, val) = (pair[0], pair[1]);
+            match key {
+                "if" => cf.set_if(val),
+                "of" => cf.set_of(val),
+                "ibs" => match get_u(val) {
+                    Some(m) => {
+                        cf.inp_block_size = m;
+                    }
+                    _ => err(format!("invalid {}: {}", key, val)),
+                },
+                "obs" => match get_u(val) {
+                    Some(m) => {
+                        cf.out_block_size = m;
+                    }
+                    _ => err(format!("invalid {}: {}", key, val)),
+                },
+                "bs" => match get_u(val) {
+                    Some(m) => {
+                        cf.inp_block_size = m;
+                        cf.out_block_size = m;
+                    }
+                    _ => err(format!("invalid {}: {}", key, val)),
+                },
+                "cbs" => match get_u(val) {
+                    Some(m) => {
+                        cf.cnv_block_size = m;
+                    }
+                    _ => err(format!("invalid {}: {}", key, val)),
+                },
+                "skip" => match get_u(val) {
+                    Some(m) => {
+                        cf.skip = m;
+                    }
+                    _ => err(format!("invalid {}: {}", key, val)),
+                },
+                "seek" => match get_i(val) {
+                    Some(m) => {
+                        cf.seek = m;
+                    }
+                    _ => err(format!("invalid {}: {}", key, val)),
+                },
+                "count" => match get_u(val) {
+                    Some(m) => {
+                        cf.count = m;
+                    }
+                    _ => err(format!("invalid {}: {}", key, val)),
+                },
+                "conv" => for elem in val.split(",") {
+                    match Conf::get_flag(elem) {
+                        Conf::INVALID_FLAG => err("invalid flag passed to conv"),
+                        flag => cf.set_flag(flag),
+                    }
+                },
+                inv => err(format!("invalid argument: {}", inv)),
+            }
+        } else {
+            err(format!("invalid argument: {}", arg));
+        }
+    }
+    cf
+}
+
+fn setup(cf: &mut Conf) {
+    let inp = if cf.is_stdin {
+        Some(InpFileWrap {
+            f: InpFileType::Stdin(io::stdin()), // TODO: use .lock() if necessary
```

Comment:
You should indeed probably use `.lock()` so you only have to lock `stdin`'s mutex once.  Same for `stdout`.